### ISSUE TITLE
"Waiting for server connection" on video upload

### ISSUE THREAD

**Issue by Subarctic-Gnarly-Trance** [owncloud/ios-legacy#187]
I'm currently having a problem with a batch of iPads using ownCloud. With some users attempting to upload a videp from the iOS app it reports "Waiting for server connection" in the uploads tab.
Occasionally it will start to upload the video and as the progress bar reaches the end this message will come up.
I am using version 3.1.2 of the iOS app and version 6.0.2 of the server.

**Comment by oc33** [comment]
@rperezb I encountered the same problem using the Owncloud app version 3.4.9 under iOS 9.3.2 against a server running Owncloud 9.0.1. The problem occurs when the video is larger than the upload and memory limits set on the server. Specifically, I could upload videos of up to 64 MB (approx 30s), while 2s videos which I estimate to be about 250 MB failed with a "Waiting for server connection" message at the end of the upload. Upload limit on the servers i 128 MB, memory limit 129 MB (shared hosting, cannot increase those). I ssh'ed into the server during the upload, but did not see any partial files in the destination folder during the upload.

Uploading a 265 MB file from a Mac using the Owncloud app worked fine (OSX 10.11.5, Owncloud app 2.2.0).

Some suggestions/questions:
- It would be really great if the Owncloud app would display file size before/during update, not only after a successful upload.
- Could the app check limits with the server before uploading and stop with a meaningful error message if the file is too large?
- I think I read somewhere that Owncloud breaks large files into smaller portions during upload. Thus, the upload file size limit of 128 MB should not be a problem. Could it be that videos are not "chopped"? Or could the problem occur due to the 129 MB memory limitation when the file is pieced together again?

**Comment by rperezb** [comment]
@oc33 😮 great feedback!

The iOS app does not take advantage of the file chunking as the desktop or the Android app does. The reason why we decided to drop this was because we'd rather the background support (keep transfers alive althought the app is closed for more than 3 min); we had to decide either support chunks or background and we though that it was better the second one so that we avoid having to have the app open for the transfer to happen.

Having said this, yes, you are right, about the 2 points:

> Some suggestions/questions:
- It would be really great if the Owncloud app would display file size before/during update, not only after a successful upload.
- Could the app check limits with the server before uploading and stop with a meaningful error message if the file is too large? 

The second point, I'd say that something pretty similar is already open, let me look for it

**Comment by oc33** [comment]
@rperezb I understand your prioritisation between backgrounding and chunking. But would it be possible to switch between the two, i.e., use chunking in the foreground for files too large to upload otherwise? But that might make user experience a bit too complicated.

**Comment by rperezb** [comment]
@oc33 this is not a trivial change, I'm afraid.
As starting point, we should and have to check the server limits, at least, in 2 sense: max uploads and availability
I have searched for the issue unsuccessfully, let me open a new one

**Comment by rperezb** [comment]
@oc33 as starting point, would it work https://github.com/owncloud/ios/issues/687 ?

**Comment by oc33** [comment]
@rperezb Yes, #687 would certainly be an improvement because the user will understand why the transfer is failing and will not waste possibly limited data volume on a failing upload. Currently, the app tries uploading every time it is activated again until one explicitly cancels the upload.

**Comment by lordenpropf** [comment]
@rperezb 
Would it be possible to add checking the  limit in upload size derived from 
_php max execution_ time and _connection speed_?

In my case the originally mentioned error "Waiting for server connection" is  caused by php_max_execution timeouts. Same symptoms - unlimited retries until the upload is cancelled manually.
I checked with my provider (shared hosting) who confirmed the error from the php logs.
In their environment (apache and php-fpm) the max_execution_time starts running on the POST-request and not on the finished file upload.

It´s not mentioned in #687 and I don´t know how difficult it is to catch those errors. 
But as the owncloud server shows the _php max execution time_ and the connection speed is known by the client (at least after the first broken upload) it should be possible to include this into the error messages.

I think without #804  **support chunked uploads** eventually quite some people in a shared environment with limited php runtime will run into timeouts eventually.

owncloud server: 9.1.4
iOS client: 3.5.3

**Comment by davivel** [comment]
Please, @lordenpropf, open a new issue for your suggestion. New ideas in closed issues are easy to be overlooked. 

Thanks.
